pkg/operation/botanist/component/kubeproxy/kube_proxy_test.go

```diff
@@ -0,0 +1,1309 @@
+    done
+immutable: true
+kind: ConfigMap
+metadata:
+  creationTimestamp: null
+  labels:
+    app: kubernetes
+    gardener.cloud/role: system-component
+    origin: gardener
+    resources.gardener.cloud/garbage-collectable-reference: "true"
+    role: proxy
+  name: ` + configMapConntrackFixScriptName + `
+  namespace: kube-system
+`
+
+			configMapCleanupScriptName = "kube-proxy-cleanup-script-a4263ada"
+			configMapCleanupScriptYAML = `apiVersion: v1
+data:
+  cleanup.sh: |
+    #!/bin/sh -e
+    OLD_KUBE_PROXY_MODE="$(cat "$1")"
+    if [ -z "${OLD_KUBE_PROXY_MODE}" ] || [ "${OLD_KUBE_PROXY_MODE}" = "${KUBE_PROXY_MODE}" ]; then
+      echo "${KUBE_PROXY_MODE}" >"$1"
+      echo "Nothing to cleanup - the mode didn't change."
+      exit 0
+    fi
+
+    /usr/local/bin/kube-proxy --v=2 --cleanup --config=/var/lib/kube-proxy-config/config.yaml --proxy-mode="${OLD_KUBE_PROXY_MODE}"
+    echo "${KUBE_PROXY_MODE}" >"$1"
+immutable: true
+kind: ConfigMap
+metadata:
+  creationTimestamp: null
+  labels:
+    app: kubernetes
+    gardener.cloud/role: system-component
+    origin: gardener
+    resources.gardener.cloud/garbage-collectable-reference: "true"
+    role: proxy
+  name: ` + configMapCleanupScriptName + `
+  namespace: kube-system
+`
+
+			podSecurityPolicyYAML = `apiVersion: policy/v1beta1
+kind: PodSecurityPolicy
+metadata:
+  creationTimestamp: null
+  name: gardener.kube-system.kube-proxy
+spec:
+  allowedCapabilities:
+  - NET_ADMIN
+  allowedHostPaths:
+  - pathPrefix: /usr/share/ca-certificates
+  - pathPrefix: /var/run/dbus/system_bus_socket
+  - pathPrefix: /lib/modules
+  - pathPrefix: /var/lib/kube-proxy
+  fsGroup:
+    rule: RunAsAny
+  hostNetwork: true
+  hostPorts:
+  - max: 10249
+    min: 10249
+  privileged: true
+  runAsUser:
+    rule: RunAsAny
+  seLinux:
+    rule: RunAsAny
+  supplementalGroups:
+    rule: RunAsAny
+  volumes:
+  - hostPath
+  - secret
+  - configMap
+  - projected
+`
+
+			clusterRolePSPYAML = `apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  creationTimestamp: null
+  name: gardener.cloud:psp:kube-system:kube-proxy
+rules:
+- apiGroups:
+  - policy
+  - extensions
+  resourceNames:
+  - gardener.kube-system.kube-proxy
+  resources:
+  - podsecuritypolicies
+  verbs:
+  - use
+`
+
+			roleBindingPSPYAML = `apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  annotations:
+    resources.gardener.cloud/delete-on-invalid-update: "true"
+  creationTimestamp: null
+  name: gardener.cloud:psp:kube-proxy
+  namespace: kube-system
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: gardener.cloud:psp:kube-system:kube-proxy
+subjects:
+- kind: ServiceAccount
+  name: kube-proxy
+  namespace: kube-system
+`
+
+			daemonSetNameFor = func(pool WorkerPool) string {
+				return "kube-proxy-" + pool.Name + "-v" + pool.KubernetesVersion
+			}
+			daemonSetYAMLFor = func(pool WorkerPool, ipvsEnabled, vpaEnabled bool) string {
```

Comment:
Should we use golang struct here? As we discussed while the refactoring task of node-local-dns, we are kind of exploiting the fact that if there are only 2-3 elements in the map then the order of elements retrieval from maps remains the same.

Reply:
Similar to most other components, I think we can adopt it once it becomes relevant.

Reply:
My idea was we shouldn't introduce new ones, but it's fine cause anyway this behavior is not going to change.